CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+7. PR will be merged.
+
+Notes:
+
+* All code needs to go through peer review. Committers may not merge their own PR.
+* PRs should **never be squashed or rebased**, even when merging. Keeping the history is important for tracking motivation behind changes later.
+
+## Compatibility
+
+All code in Requests must be compatible with PHP 5.6.
+Requests is used in WordPress, and hence is tied to [its compatibility][wp-requirements].
+
+Requests is also bound to not break backwards compatibility.
+In semantic versioning terms, this means we will be in the 2.x release cycle for the forseeable future.
+
+Generally speaking, the goal of Requests is to provide an interface that smoothes the differences across various server setups.
+That means there shouldn't be any code relying on any extensions (even if they're common) unless we also provide a pure-PHP version (that is, we can use extensions to improve performance when available). Only two extensions are currently used: SPL, and cURL.
+SPL is always available from PHP 5.3 onwards.
+cURL is only used for better performance over sockets.
+
+[wp-requirements]: https://wordpress.org/about/requirements/
+
+
+## Coding Style
+
+Please follow the existing coding style. This project uses [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer)
+to detect coding standard violations and apply automated fixes (whenever possible).  
+All files can be checked for coding standard violations by running `composer run checkcs`.
+Any automatically applicable fixes can be applied by running `composer run fixcs`.
+
+
+## Unit Tests
+
+PRs should include unit tests for any changes.
+These are written in PHPUnit, and should be added to the file corresponding to the class they test (that is, tests for `library/Requests/Cookie.php` would be in `tests/Cookie.php`).
+
+Where possible, features must be unit tested.
+We aim for >90% coverage at all times.
+The master branch may drop below 90% if features are merged independently of their tests, but there is a hard limit of 85%. Release versions must have >90% coverage.
+
+For complex features by third-parties, PRs may be merged that drop coverage below the 90% threshold, with the intent of increasing tests back up in a subsequent PR.
+
+### Prerequisites
+- PHP >= 5.6
+- Composer
+- Python 3
+- pip3
+- mitmproxy (`pip3 install mitmproxy`)
+
+### Running the Tests
+```bash
+# Start the test server
+PORT=8080 vendor/bin/start.sh
+export "REQUESTS_TEST_HOST_HTTP=localhost:8080"
+
+# Start the proxy server
+PORT=9002 tests/utils/proxy/start.sh
+PORT=9003 AUTH="test:pass" tests/utils/proxy/start.sh
+export "REQUESTS_HTTP_PROXY=localhost:9002"
+export "REQUESTS_HTTP_PROXY_AUTH=localhost:9003"
+export "REQUESTS_HTTP_PROXY_AUTH_USER=test"
+export "REQUESTS_HTTP_PROXY_AUTH_PASS=pass"
+
+# Run the tests
+composer test
+
+# Stop the proxy server
+PORT=9002 tests/utils/proxy/stop.sh
+PORT=9003 tests/utils/proxy/stop.sh
+
+# Stop the test server
+vendor/bin/stop.sh
+```
```

Comment:
Suggestion: would it be an idea to save this as a script in a `bin` directory and to direct users to run that script instead ?

Question: will this work on Windows (considering this is a bash script) ?